server/socketManager.js

```diff
@@ -252,6 +221,211 @@ class SocketManager extends Server {
       const resp = await client.call(event, ...args);
       return resp;
     });
+
+    // listener for tcp proxy requests (useful for in-browser nodes)
+    socket.bind('tcp connect', (port, host) => {
+      this.handleTCPConnect(socket, port, host);
+    });
+  }
+
+  handleTCPConnect(ws, port, host) {
+    const state = this.sockets.get(ws);
+    assert(state);
+    if (state.socket) {
+      this.proxyLog('info', 'Client is trying to reconnect (%s).', state.host);
+      return;
+    }
+
+    if (
+      (port & 0xffff) !== port ||
+      typeof host !== 'string' ||
+      host.length === 0
+    ) {
+      this.proxyLog('error', 'Client gave bad arguments (%s).', state.host);
+      ws.fire('tcp close');
+      ws.destroy();
+      return;
+    }
+
+    let raw, addr;
+    try {
+      raw = IP.toBuffer(host);
+      addr = IP.toString(raw);
+    } catch (e) {
+      this.proxyLog(
```

Comment:
Do you like the way the linter separates things here?

Reply:
heh. Don't love it but doesn't bother me too much. The worst to me is the React text where it adds a `{' '}` to add a space and a new line 🤦‍♂️